### PR TITLE
Make ArgumentCollection.GetArgumentNames return type IEnumerable<string?>

### LINKED ISSUE #1922: ArgumentCollection.ArgumentNames has type IEnumerable<string> but may contain nulls
In F# (at least), it is possible to declare a method with anonymous parameters.
In this case, `ArgumentCollection.ArgumentNames` will contain `null`s, but we do not indicate that the elements are nullable strings.

### PR DESCRIPTION
**Pull request by blairconrad**
Fixes #1922

### CODE REVIEW

**Comment by blairconrad** [comment]
I added a bonus commit to clean up some "hints". Happy to drop if it offends.

File: src/FakeItEasy/Configuration/ArgumentCollection.cs [hunk 1]
```diff
@@ -54,7 +53,7 @@ public int Count
         /// <summary>
         ///   Gets the names of the arguments in the list.
```
Comment (blairconrad) [inline src/FakeItEasy/Configuration/ArgumentCollection.cs:54]: Ugh. There's an opportunity to add docs here around when the names might be null. I dropped the ball. Stay tuned.

**Comment by blairconrad** [comment]
Docs added!

**Review by thomaslevesque — APPROVED** [review]
Perfect, thanks @blairconrad!

**Comment by blairconrad** [comment]
Thanks, @thomaslevesque.

**Pull request merged**